data/templates/set_bmc_credentials.sh

```diff
@@ -1,9 +1,27 @@
 #!/usr/bin/env bash
 
+channel=''
+function set_channel()
+{
+    for i in {1..15}; do
+        ipmitool user list $i &>/dev/null
+        status=$?
+        if [ "$status" -eq "0" ] ; then
+            channel=$i
+            break
+        fi
+    done
+}
+set_channel
+echo "channel number is" $channel
+if [ -z "${channel}" ]; then
+ echo "Channel number was not set correctly, exiting script"
+exit
```

Comment:
You may want to do a `exit 1` if this is an error case.  In case the caller wants to evaluate the script exit status $?.

fyi: http://tldp.org/LDP/abs/html/exitcodes.html